### PR TITLE
CSP-667-Check your answers-Region

### PR DESCRIPTION
**Pull request by YuvrajPatil-DFE**

### CODE REVIEW

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/RegionsControllerTests/RegionsControllerGetTests.cs [hunk 1]
```diff
@@ -39,8 +39,8 @@ public async Task Get_ViewResult_HasCorrectViewPath([Greedy] RegionsController s
         OnboardingSessionModel sessionModel,
         [Greedy] RegionsController sut)
     {
-        sut.AddUrlHelperMock().AddUrlForRoute(RouteNames.Onboarding.CurrentJobTitle);
         sessionServiceMock.Setup(s => s.Get<OnboardingSessionModel>()).Returns(sessionModel);
+        sut.AddUrlHelperMock().AddUrlForRoute(sessionModel.HasSeenPreview ? RouteNames.Onboarding.CheckYourAnswers : RouteNames.Onboarding.CurrentJobTitle);
```
Comment (DevenKShah) [inline src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/RegionsControllerTests/RegionsControllerGetTests.cs:43]: We need to add specific tests for Backlink for both case where HasSeenPreview is true and false.

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Models/CheckYourAnswersViewModelTests.cs [hunk 2]
```diff
@@ -38,4 +38,39 @@ public void Ctor_SetsJobTitle()
...
     }
+
+    [Test]
+    public void Ctor_SetsRegionChangeLink()
+    {
+        var expectedRegion = 101;
+        var expectedJobTitle = "Some Title";
+
+        Mock<IUrlHelper> mockUrlHelper = new();
+        mockUrlHelper.AddUrlForRoute(RouteNames.Onboarding.Regions);
```
Comment (DevenKShah) [inline src/SFA.DAS.ApprenticeAan.Web.UnitTests/Models/CheckYourAnswersViewModelTests.cs:49]: Need to override default url here

File: src/SFA.DAS.ApprenticeAan.Web/Controllers/Onboarding/RegionsController.cs [hunk 3]
```diff
@@ -54,16 +54,17 @@ public async Task<IActionResult> Post(RegionsSubmitModel submitmodel)
         }
 
         sessionModel.RegionId = submitmodel.SelectedRegionId;
+        sessionModel.RegionName = model.Regions.First(x => x.Id == sessionModel.RegionId).Area;
         _sessionService.Set(sessionModel);
 
         return RedirectToRoute(RouteNames.Onboarding.ReasonToJoin);
```
Comment (DevenKShah) [inline src/SFA.DAS.ApprenticeAan.Web/Controllers/Onboarding/RegionsController.cs:60]: On post we need to navigate back to CheckYourAnswers if HasSeenPreview is true

**Review by DevenKShah — CHANGES_REQUESTED** [review]

**Review by MarkFCain — COMMENTED** [review]
Please add coverage asked for, I don't see any other issues

**Review by MarkFCain — APPROVED** [review]

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/RegionsControllerTests/RegionsControllerPostTests.cs [hunk 4]
```diff
@@ -73,4 +91,35 @@ public class RegionsControllerPostTests
         result.As<RedirectToRouteResult>().Should().NotBeNull();
         result.As<RedirectToRouteResult>().RouteName.Should().Be(RouteNames.Onboarding.ReasonToJoin);
     }
+
+    [MoqAutoData]
+    public async Task Post_Regions_RedirectsToCheckYourAnswers(
```
Comment (DevenKShah) [inline src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/RegionsControllerTests/RegionsControllerPostTests.cs:96]: Post_IsValidAndHasSeenPreview_RedirectsToCheckYourAnswers

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/RegionsControllerTests/RegionsControllerGetTests.cs [hunk 5]
```diff
@@ -34,17 +34,49 @@ public async Task Get_ViewResult_HasCorrectViewPath([Greedy] RegionsController s
...
+        sessionModel.HasSeenPreview = true;
+        sessionServiceMock.Setup(s => s.Get<OnboardingSessionModel>()).Returns(sessionModel);
+
+        var result = await sut.Get();
+
+        result.As<ViewResult>().Model.As<RegionsViewModel>().BackLink.Should().Be(checkYourAnswersUrl);
+    }
+
+    [MoqAutoData]
+    public async Task Get_ViewModelHasSeenPreviewIsFalse_RedirectsRouteToCurrentJobTitle(
```
[the remark this answers is not in the document]
Comment (YuvrajPatil-DFE) [reply src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/RegionsControllerTests/RegionsControllerGetTests.cs:53]: Renamed to Get_ViewModelHasSeenPreviewIsFalse_BankLinkSetsToToCurrentJobTitle

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/RegionsControllerTests/RegionsControllerGetTests.cs [hunk 6]
```diff
@@ -34,17 +34,49 @@ public async Task Get_ViewResult_HasCorrectViewPath([Greedy] RegionsController s
     }
 
     [MoqAutoData]
-    public async Task Get_ViewModel_GetsRegionsViewModel(
+    public async Task Get_ViewModelHasSeenPreviewIsTrue_RedirectsRouteToCheckYourAnswers(
```
[the remark this answers is not in the document]
Comment (YuvrajPatil-DFE) [reply src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/RegionsControllerTests/RegionsControllerGetTests.cs:37]: Get_ViewModelHasSeenPreviewIsTrue_BankLinkSetsToCheckYourAnswers

**Review by DevenKShah — DISMISSED** [review]

**Review by DevenKShah — APPROVED** [review]

**Pull request merged**